### ISSUE TITLE
Hello, Thanks for your work!Why didn't you load the weights pretrained on ImageNet for the resnet18?

### ISSUE THREAD

**Issue by yang-gis** [wvangansbeke/Unsupervised-Classification#13]

**Comment by yang-gis** [comment]
Your code is "get_model(P)" instead of  "get_model(P, pretrained_weights_path)"

**Comment by SimonVandenhende** [comment]
I am not sure if I completely understand you question. But I assume you are asking why we don't load the ImageNet pre-trained weights from torchvision for the ResNet-18 model when performing simclr? Notice that these weights were obtained by training a model in a fully-supervised manner on ImageNet. In this case, we assume that the labels are unavailable. As a consequence,  obtaining the pre-trained weights from PyTorch is not possible.

**Comment by yang-gis** [comment]
> I am not sure if I completely understand you question. But I assume you are asking why we don't load the ImageNet pre-trained weights from torchvision for the ResNet-18 model when performing simclr? Notice that these weights were obtained by training a model in a fully-supervised manner on ImageNet. In this case, we assume that the labels are unavailable. As a consequence, obtaining the pre-trained weights from PyTorch is not possible.

Thanks for your reply!I think it's neccesary for small dataset feature learning to load the ImageNet pre-trained weights from torchvision for the ResNet-18 model .

**Comment by SimonVandenhende** [comment]
All results in the paper were obtained by initializing the network weights from scratch. 
On CIFAR-10, STL-10 and CIFAR-100-20, the network weights are initialized by running simclr.py
On ImageNet, the network weights are initialized by running moco.py
Both of these schemes operate in a self-supervised manner. 
When performing scan.py we transfer the weights from the pretext task (simclr or moco). When performing selflabel.py, we transfer the weights obtained after running scan.py

**Comment by wvangansbeke** [comment]
It's unsupervised (actually self-supervised), so you assume you don't have labels for ImageNet. I'm closing this for now.

**Comment by yang-gis** [comment]
> All results in the paper were obtained by initializing the network weights randomly before training the pretext task.
> On CIFAR-10, STL-10 and CIFAR-100-20, the network weights are initialized by running simclr.py
> On ImageNet, the network weights are initialized by running moco.py
> Both of these schemes operate in a self-supervised manner.
> When performing scan.py we transfer the weights from the pretext task (simclr or moco). When performing selflabel.py, we transfer the weights obtained after running scan.py

oh,I want to use my own small dataset(about 1000 images, 10 class).If  I train the pretext task on my own dataset , how to do with your code?Please give me some help.

**Comment by wvangansbeke** [comment]
Have a look at the file `moco.py`, the pretrained weights are downloaded automatically. MoCo uses instance discrimination to pretrain without labels. We start from these weights for ImageNet. They only include weights for a resnet-50 though.
